network/peer.go

```diff
@@ -836,3 +879,45 @@ func (p *peer) getIP() utils.IPDesc {
 	defer p.ipLock.RUnlock()
 	return p.ip
 }
+
+// addAlias marks that we have found another
+// IP that we can connect to this peer at.
+//
+// assumes stateLock is held
+func (p *peer) addAlias(ip utils.IPDesc) {
+	p.ipLock.Lock()
+	defer p.ipLock.Unlock()
+
+	p.net.peerAliasIPs[ip.String()] = struct{}{}
```

Comment:
ipLock is on the peer.   but you are updating the network.peerAliasIPs..  
The lock needs to be held in network to safely do the update, or you might get a panic.

Reply:
Yep! `addAlias` is called when `stateLock` (https://github.com/ava-labs/avalanchego/pull/746/files#diff-96a7f5c352cd3d2d1ee3c7c55ed95e668c6edb28a4fa337e6c60270471bdab61R1176-R1177) is already held: https://github.com/ava-labs/avalanchego/pull/746/files#diff-96a7f5c352cd3d2d1ee3c7c55ed95e668c6edb28a4fa337e6c60270471bdab61R1212

The comments indicate that `stateLock` must be held when called (like some other functions on `peer`): https://github.com/ava-labs/avalanchego/pull/746/files#diff-b06a14805cb9806d82ef18d987b32228ab9b36a4e679c7c128efd711038991b2R886